From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Laura Abbott <labbott@redhat.com>
Date: Thu, 2 Jan 2020 14:24:46 -0500
Subject: [PATCH] mptsas: Taint kernel if mptsas is loaded

Author: Tomas Henzl <thenzl@redhat.com>
Date:   Thu Jan 31 16:54:50 2019 -0500

    [message] mptsas: Taint kernel if mptsas is loaded

    Message-id: <20190131165452.7562-2-thenzl@redhat.com>
    Patchwork-id: 239764
    O-Subject: [RHEL8 e-stor 1/3] mptsas: Taint kernel if mptsas is loaded
    Bugzilla: 1666730
    RH-Acked-by: Jarod Wilson <jarod@redhat.com>
    RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
    RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>

    RHEL-only

    We are only keeping it to ease developer transitions, we're not
    supporting it, so mark it as unsupported by tainting the kernel and
    issuing an impossible to misunderstand message to the log

    Signed-off-by: Herton R. Krzesinski <herton@redhat.com>

Upstream Status: RHEL only
Signed-off-by: Laura Abbott <labbott@redhat.com>
---
 drivers/message/fusion/mptsas.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/message/fusion/mptsas.c b/drivers/message/fusion/mptsas.c
index 6a79cd0ebe2b..c17614cd6ed8 100644
--- a/drivers/message/fusion/mptsas.c
+++ b/drivers/message/fusion/mptsas.c
@@ -5290,6 +5290,11 @@ mptsas_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 		    ioc, MPI_SAS_OP_CLEAR_ALL_PERSISTENT);
 	}

+#ifdef CONFIG_RHEL_DIFFERENCES
+	add_taint(TAINT_SUPPORT_REMOVED, LOCKDEP_STILL_OK);
+	pr_warn("MPTSAS MODULE IS NOT SUPPORTED\n");
+#endif
+
 	error = scsi_add_host(sh, &ioc->pcidev->dev);
 	if (error) {
 		dprintk(ioc, printk(MYIOC_s_ERR_FMT
-- 
2.26.2

